lua/code_action_menu/lsp_objects/edits/text_document_edit.lua

```diff
@@ -2,6 +2,40 @@ local TextDocumentEditStatusEnum = require(
   'code_action_menu.enumerations.text_document_edit_status_enum'
 )
 
+local function get_line(uri, row)
+  local uv = vim.loop
+  -- load the buffer if this is not a file uri
+  -- Custom language server protocol extensions can result in servers sending URIs with custom schemes. Plugins are able to load these via `BufReadCmd` autocmds.
+  if uri:sub(1, 4) ~= "file" then
+    local bufnr = vim.uri_to_bufnr(uri)
+    vim.fn.bufload(bufnr)
+    return (vim.api.nvim_buf_get_lines(bufnr, row, row + 1, false) or { "" })[1]
+  end
+
+  local filename = vim.uri_to_fname(uri)
+
+  -- use loaded buffers if available
+  if vim.fn.bufloaded(filename) == 1 then
+    local bufnr = vim.fn.bufnr(filename, false)
+    return (vim.api.nvim_buf_get_lines(bufnr, row, row + 1, false) or { "" })[1]
+  end
+
+  local fd = uv.fs_open(filename, "r", 438)
+  -- TODO: what should we do in this case?
+  if not fd then return "" end
+  local stat = uv.fs_fstat(fd)
+  local data = uv.fs_read(fd, stat.size, 0)
+  uv.fs_close(fd)
+
+  local lnum = 0
```

Comment:
Nitpick for code quality: code gets usually written once (in its final iteration) and you have the support of code completion. But it gets read many times over and over again. So please use full expressing variable names that clearly state what it is. It sometimes is a hard balance of not making them too long and rely on a special context within this variable lives. But `lnum` to `line_number` is really very short and nice to read (if that naming fits what you intend to express, see the guessing game has started).
As recommendation if you are interested into these topics a book recommendation is `Clean Code - A Handbook of Agile Software Craftsmanship` by Robert C. Martin. Not the holy grail, but worth to be read in combination with other to build up a sophisticated knowledge base what makes code more clean.